sympy/printing/latex.py

```diff
@@ -2054,11 +2051,26 @@ def _print_Range(self, s):
         elif s.stop.is_infinite:
             it = iter(s)
             printset = next(it), next(it), dots
-        elif len(s) > 4:
-            it = iter(s)
-            printset = next(it), next(it), dots, s[-1]
         else:
-            printset = tuple(s)
+            try:
+                if (s.size < 4) == True:
+                    printset = tuple(s)
+                else:
+                    it = iter(s)
+                    printset = next(it), next(it), dots, s[-1]
+            except (TypeError, ValueError):
```

Comment:
Rather than catch ValueError or TypeError can this have an explicit check. Also what raises these exceptions? Can we eliminate them at the source?

I don't like raising or catching either ValueError or TypeError since those exception types are regularly raised when there is a bug in the code.

Reply:
This is also related to that the symbolic symbols in `Range` must be integer for some things to work. Both `size` and getting `next` (or maybe at `iter`) will throw an exception if it cannot get anything, so it was way to deal with both of them.

The thing is that `Range(i, j)` will not work, but `Range(i, i+3)` will, so the easy way is to try and catch the exception (the alternative is to have the same tests as those throwing the exception, which then gives two places to change if this is improved). I know that it can hide bugs, but I think that in the printing code it is less of a problem.

Reply:
Can you decide based on ` is_empty `being None

Reply:
Can there not just be an `is_iterable` attribute for Range? Basically if `is_iterable` is True then it should be safe to iterate or something.

Reply:
> Can you decide based on `is_empty`being None

It is probably possible to get rid of one of the exceptions then (the one coming from `size`). I do not think it existed when I wrote this code though. :-) (At least not in master.)

Reply:
One exception removed now. Would have to do something like `is_iterable` to get rid of the other one. However, `Range` defines it to be `True` already. Should I replace it with a function checking if it actually can be iterated?